pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -195,3 +197,43 @@ func (c *ClusterController) updateClusterCephVersion(image string, cephVersion c
 		return
 	}
 }
+
+// updateCephStorage queries the device class information and udates the CR
+func (c *ClusterController) updateCephStorage() {
+	var err error
+	var devices []string
+	retryCount := 5
+	cephCluster := &cephv1.CephCluster{}
+	cephClusterStorage := cephv1.CephStorage{}
+
+	//wait a bit longer to get osd details reflected in the crush map
+	for i := 0; i < retryCount; i++ {
+		devices, err = cephclient.GetDeviceClasses(c.context, c.namespacedName.Namespace)
+		if err != nil {
+			logger.Errorf("failed to get storage device classes. %v", err)
+			return
+		}
+		if len(devices) > 0 {
+			break
+		}
+		time.Sleep(checkStorageWaitTime)
+	}
```

Comment:
This is to avoid updating with empty device classes in the CR, as the details on device classes is taking around 10 to 15 secs to reflect in the `crush`.